src/dns-test.js

```diff
@@ -293,19 +341,43 @@ async function sendQueries(nameservers_ipv4) {
  */
 function sendTelemetry(payload) {
     payload.measurementID = measurementID;
-    browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+    console.log(payload);
+    // browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+}
+
+async function fetchTest() {
+    // Unsure if we want to wrap the fetch in a try/catch. This would enable 
+    // us to handle network errors, in case that matters to us. I don't think 
+    // it does, though. I think all we want is to stop the measurement from 
+    // running if we don't get the response we expect. Also, if the fetch fails, 
+    // it doesn't make sense to send anything to telemetry, because that 
```

Comment:
This comment is inaccurate. Telemetry pings are queued until they can be sent.

See https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/concepts/submission.html

Reply:
Gotcha; thanks. I'll check with ekr whether we care to queue a ping for this error.

Reply:
We decided to queue up some telemetry when we infer that the client doesn't have connectivity, i.e. via the captivePortal API and a fetch() test with a webpage we control. Addressed by 50e06b7.